src/System/DataLogger.cpp

```diff
@@ -0,0 +1,232 @@
+#include "DataLogger.h"
+#include "settings.h"
+
+using namespace std;
+
+//public methods:
+
+DataLogger::DataLogger(int pinNumber, bool osync /*= false*/) {
+    if (SD.begin(pinNumber, SPI1)) {
```

Comment:
I just noticed this.  All system begin / init methods should be called from setup.

DataLogger is going to be a global object, so its constructor will be called during this program's global initialization.  Generally, you want to make sure that all system resources are initialized before you make any system calls, so you should move this code to an init method which we can call from the main setup function.